types/mdx/md.d.ts

```diff
@@ -0,0 +1,9 @@
+///<reference path="./index.d.ts"/>
+
+/**
+ * An markdown file which exports a JSX component.
+ */
+declare module '*.md' {
+    import mdx = require('*.mdx');
+    export = mdx;
+}
```

Comment:
I wanted to make this opt-in, because I imagine some people may want to use markdown files differently. I.e. use `.mdx` files using `@mdx-js/loader` and `.md` files using `file-loader`. This is highly subjective though. If you have strong feelings for merging them into `index.d.ts` that’s fine with me. The same goes for other markdown extensions.

Another option would be to not support `.md` at all here.

--- 

> This code seems to mean that, when `.mdx` files are augmented with more exports, those exports also apply to `.md` files. Is that correct, intentional, and should it be that way?

This is intentional, but also subjective.